eclipse-theia/theia
# Replace icons with codicons

#### What it does

Closes #6115

Does a lot of css work:

* Replaces all internally used icons (fontawesome, custom svg files) with codicons
* Adds a new utility method for codicons, namely `codicon` and `codiconArray`.
* Aligns button-icon css with vscode: Buttons in toolbars, etc. are now highlighted with a backdrop

I decided to leave the old svg icons in the codebase for backwards compatibility. I also couldn't notice any breaking change, but please keep an eye out for them.


#### How to test

Before starting to test I would recommend to start a second Theia instance to compare visual changes. The following testing steps are only the most important parts that have changed. As every icon in the app has changed, please refer to the code to gather a list of everything that has changed. Also, change the icon theme to `seti` at appropriate times to check for tab-bar items, etc.

* Check toolbar icons, every icon should show a backdrop on hover.
* Check tabbar close icons, every close icon should show a backdrop on hover.
* Check status bar icons
* Check notification widget icons
* Check debug view widget icons
* Check scm view widget icons
* Check problems view widget icons
* Check extensions view widget icons

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
